src/pages/account/settings/index.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+
+import { Dispatch } from 'redux';
+import { FormattedMessage } from 'umi-plugin-react/locale';
+import { GridContent } from '@ant-design/pro-layout';
+import { Menu } from 'antd';
+import { connect } from 'dva';
+import BaseView from './components/base';
+import BindingView from './components/binding';
+import { CurrentUser } from './data.d';
+import NotificationView from './components/notification';
+import SecurityView from './components/security';
+import styles from './style.less';
+
+const { Item } = Menu;
+
+interface SettingsProps {
+  dispatch: Dispatch<any>;
+  currentUser: CurrentUser;
+}
+
+type SettingsStateKeys = 'base' | 'security' | 'binding' | 'notification';
+interface SettingsState {
+  mode: 'inline' | 'horizontal';
+  menuMap: {
+    [key: string]: React.ReactNode;
+  };
+  selectKey: SettingsStateKeys;
+}
+@connect(({ accountAndsettings }: { accountAndsettings: { currentUser: CurrentUser } }) => ({
+  currentUser: accountAndsettings.currentUser,
+}))
+class Settings extends Component<SettingsProps, SettingsState> {
+  main: HTMLDivElement | undefined = undefined;
+
+  constructor(props: SettingsProps) {
+    super(props);
+    const menuMap = {
+      base: (
+        <FormattedMessage id="accountandsettings.menuMap.basic" defaultMessage="Basic Settings" />
+      ),
+      security: (
+        <FormattedMessage
+          id="accountandsettings.menuMap.security"
+          defaultMessage="Security Settings"
+        />
+      ),
+      binding: (
+        <FormattedMessage
+          id="accountandsettings.menuMap.binding"
+          defaultMessage="Account Binding"
+        />
+      ),
+      notification: (
+        <FormattedMessage
+          id="accountandsettings.menuMap.notification"
+          defaultMessage="New Message Notification"
+        />
+      ),
+    };
+    this.state = {
+      mode: 'inline',
+      menuMap,
+      selectKey: 'base',
+    };
+  }
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'accountAndsettings/fetchCurrent',
+    });
+    window.addEventListener('resize', this.resize);
+    this.resize();
+  }
+
+  componentWillUnmount() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'componentWillUnmount' to have a typedef](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)